### PR TITLE
discovery/dns: add test case for SDConfig.UnmarshalYAML.

### PR DESCRIPTION
**Pull request by johncming**
No test code coverage:

https://github.com/prometheus/prometheus/blob/4f47806a7d33507dc9ff89c96cc41c91d3fd1d77/discovery/dns/dns.go#L74-L94

### CODE REVIEW

File: discovery/dns/dns_test.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ package dns
 import (
 	"context"
 	"fmt"
+	"gopkg.in/yaml.v2"
```
Comment (simonpasquier) [inline discovery/dns/dns_test.go]: could you move it the second group of imports (eg `github.com/go-kit/kit/log`)?

File: discovery/dns/dns_test.go [hunk 2]
```diff
@@ -178,3 +179,96 @@ func TestDNS(t *testing.T) {
...
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			var config SDConfig
+			d := marshal(c.input)
+			err := config.UnmarshalYAML(unmarshal(d))
+			if (err != nil) != c.expectErr {
+				t.Errorf("test case %q: UnmarshalYAML expect error: %v, got: %v; %v", c.name, c.expectErr, err != nil, err)
```
Comment (simonpasquier) [inline discovery/dns/dns_test.go]: no need to repeat `c.name` in the error message since it is already passed to `t.Run`.

Comment (johncming) [reply discovery/dns/dns_test.go]: ok.

File: discovery/dns/dns_test.go (hunk 1 above)
Comment (johncming) [reply discovery/dns/dns_test.go]: ok.

**Comment by johncming** [comment]
> Thanks! A few nits otherwise LGTM.

updated.

File: discovery/dns/dns_test.go [hunk 3]
```diff
@@ -23,9 +23,9 @@ import (
 	"github.com/go-kit/kit/log"
 	"github.com/miekg/dns"
 	"github.com/prometheus/common/model"
-
 	"github.com/prometheus/prometheus/discovery/targetgroup"
 	"github.com/prometheus/prometheus/util/testutil"
+	"gopkg.in/yaml.v2"
```
Comment (simonpasquier) [inline discovery/dns/dns_test.go]: nit but it should be listed in this group:

```
	"github.com/go-kit/kit/log"
	"github.com/miekg/dns"
	"github.com/prometheus/common/model"
```

Comment (simonpasquier) [reply discovery/dns/dns_test.go]: the rule is

```
import (
  <stdlib packages>

  <3rd party packages>

  <github.com/prometheus/prometheus packages>
)
```

Comment (johncming) [reply discovery/dns/dns_test.go]: ok

File: discovery/dns/dns_test.go [hunk 4]
```diff
@@ -178,3 +179,94 @@ func TestDNS(t *testing.T) {
 		})
 	}
 }
+
+func TestSDConfigUnmarshalYAML(t *testing.T) {
+
```
Comment (cstyan) [inline discovery/dns/dns_test.go]: nit, don't need newline here

Comment (johncming) [reply discovery/dns/dns_test.go]: ok.

**Comment by johncming** [comment]
updated.

**Comment by simonpasquier** [comment]
Thanks!

**Pull request merged**